src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -1063,13 +1104,111 @@ void PerformInteractionRequest::SendBothModeResponse(
       msg_param.empty() ? NULL : &msg_param;
   std::string info = app_mngr::commands::MergeInfos(
       ui_perform_info, ui_info_, vr_perform_info, vr_info_);
+
   DisablePerformInteraction();
+
   SendResponse(result,
                perform_interaction_result_code,
                info.empty() ? NULL : info.c_str(),
                response_params);
 }
 
+mobile_apis::Result::eType
+PerformInteractionRequest::PrepareResultCodeForResponse(
+    const app_mngr::commands::ResponseInfo& ui_response,
+    const app_mngr::commands::ResponseInfo& vr_response) {
+  LOG4CXX_DEBUG(
+      logger_, "InteractionMode = " << static_cast<int32_t>(interaction_mode_));
+
+  auto result = mobile_api::Result::INVALID_ENUM;
+  switch (interaction_mode_) {
+    case mobile_apis::InteractionMode::VR_ONLY: {
+      result = MessageHelper::HMIToMobileResult(vr_result_code_);
+      break;
+    }
+
+    case mobile_apis::InteractionMode::MANUAL_ONLY: {
+      result = MessageHelper::HMIToMobileResult(ui_result_code_);
+      break;
+    }
+
+    case mobile_apis::InteractionMode::BOTH: {
+      if (IsVRPerformInteractionResponseSuccessfulInBothMode()) {
```

Comment:
VR shouldn't take precedent here. This condition should just be dependent on which response included a valid choice_id.

```
if VR response had valid choice_id:
    return vr_code
else if UI response had valid choice_id:
    return ui_code
else
    return CommandRequestImpl::Prepare(ui, vr)
```

`IsVRPerformInteractionResponseSuccessfulInBothMode()` can be removed.